tests/spread/plugins/v2/snaps/colcon-ros2-foxy-rlcpp-hello/src/hello.cpp

```diff
@@ -0,0 +1,28 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include "rclcpp/rclcpp.hpp"
+
+int main(int argc, char * argv[])
+{
+  rclcpp::init(argc, argv);
+  rclcpp::executors::SingleThreadedExecutor exec;
+  rclcpp::NodeOptions options;
+
+  printf("hello world");
```

Comment:
This barely uses ROS. The talker/listener example was far more realistic, in my opinion.

Reply:
I sort of figured if the rlcpp init succeeded, it'd be adequate.  I don't think we're really stressing ROS here so much as the plugin/extension worked fine, and I thought this was sufficient to cause verify rlcpp linkage.

The only requirement we really have is it has to say "hello world" and exit without fussing about, so it's consistent with the remaining v2 plugin tests which cover other parts of the lifecycle (clean, rebuild, etc.). In theory I was supposed to be able to write a node loop that did exactly that then exited, but it appeared buggy or my interpretation of the docs was wrong.

That said, I agree with your point, if we want to do more stressing of ROS, I think that's great and we should certainly add more tests outside of this task (or figure out how to make it work).

Reply:
Are you saying that the existing talker/listener demo that exits after it receives something wasn't working properly? Yeah happy to help with that. We also want to add more spread tests for all the supported ROS distros, so it seems fine to leave this for now and tackle "better ROS tests" as a general thing.